### PR TITLE
feat(CAL-04) : Adding database engine

### PR DESCRIPTION
**Pull request by gabrielsperche**

### CODE REVIEW

File: src/main/java/org/splitec/client/DatabaseClient.java [hunk 1]
```diff
@@ -0,0 +1,85 @@
...
+import com.azure.cosmos.models.CosmosContainerResponse;
+import com.azure.cosmos.models.CosmosDatabaseResponse;
+import  org.splitec.config.CosmosConfig;
+import java.util.ArrayList;
+
+public class DatabaseClient {
+
+    private CosmosClient client;
+
+    private final String databaseName = "CalamaDB";
```
Comment (gabrielsperche) [inline src/main/java/org/splitec/client/DatabaseClient.java:18]: @luucasfl Please add this a constant

File: src/main/java/org/splitec/client/DatabaseClient.java [hunk 2]
```diff
@@ -0,0 +1,85 @@
...
+    }
+
+
+    public CosmosContainer getContainer() {
+        return container;
+    }
+
+    public void buildClient() {
+        try {
+            System.out.println("Using Azure Cosmos DB endpoint: " + CosmosConfig.HOST);
```
Comment (gabrielsperche) [inline src/main/java/org/splitec/client/DatabaseClient.java:39]: @luucasfl ?????????????????????????????????????????????????????????????????????? why are you loggin that in the console? please implement a logger class

File: src/main/java/org/splitec/client/DatabaseClient.java [hunk 3]
```diff
@@ -0,0 +1,85 @@
...
+
+            ArrayList<String> preferredRegions = new ArrayList<>();
+            preferredRegions.add("West US");
```
Comment (gabrielsperche) [inline src/main/java/org/splitec/client/DatabaseClient.java:42]: @luucasfl Constant

File: src/main/java/org/splitec/client/DatabaseClient.java [hunk 4]
```diff
@@ -0,0 +1,85 @@
...
+                    .preferredRegions(preferredRegions)
+                    .userAgentSuffix("CosmosDBJavaQuickstart")
+                    .consistencyLevel(ConsistencyLevel.EVENTUAL)
+                    .buildClient();
+
+            createDatabaseIfNotExists();
+            createContainerIfNotExists();
+        }
+        catch (Exception e) {
+            e.printStackTrace();
```
Comment (gabrielsperche) [inline src/main/java/org/splitec/client/DatabaseClient.java:56]: @luucasfl ?????????????????????????????????????????????????????????????W WHY WE ARE SWALLOWING THE EXCEPTION HERE?

File: src/main/java/org/splitec/client/DatabaseClient.java [hunk 5]
```diff
@@ -0,0 +1,85 @@
...
+            System.err.printf("Cosmos getStarted failed with %s%n", e);
```
Comment (gabrielsperche) [inline src/main/java/org/splitec/client/DatabaseClient.java:57]: @luucasfl Please, NEVER LOG A ERROR MESSAGE IN CONSOLE, THAT A BAD PRACTICE

File: src/main/java/org/splitec/client/DatabaseClient.java [hunk 6]
```diff
@@ -0,0 +1,85 @@
...
+        }
+    }
+
+    private void createDatabaseIfNotExists() {
+        System.out.println("Create database " + databaseName + " if not exists.");
+
+        //  Create database if not exists
+        CosmosDatabaseResponse databaseResponse = client.createDatabaseIfNotExists(databaseName);
```
Comment (gabrielsperche) [inline src/main/java/org/splitec/client/DatabaseClient.java:65]: @luucasfl WHY WE ARE CHECKING IF THE DATABASE DOES NOT EXIST? PLEASE EXPLAIN JUST ONE SCENARIO THAT WE WILL NEED TO CREATE A DATABASE IN RUNTIME

File: src/main/java/org/splitec/service/CosmosService.java [hunk 7]
```diff
@@ -0,0 +1,49 @@
+package org.splitec.service;
+
+import com.azure.cosmos.CosmosContainer;
+import com.azure.cosmos.models.CosmosQueryRequestOptions;
+import com.azure.cosmos.util.CosmosPagedIterable;
+import org.splitec.client.DatabaseClient;
+import org.splitec.model.User;
+
+import java.util.concurrent.atomic.AtomicReference;
+
```
Comment (gabrielsperche) [inline src/main/java/org/splitec/service/CosmosService.java:10]: @luucasfl Please add the annotation to Service bean

File: src/main/java/org/splitec/service/UvHealthService.java [hunk 8]
```diff
@@ -43,13 +47,10 @@ public int findSecureExposureTime(GetIndex.SafeExposureTime exposureTimeResponse
...
 
-  public int findUserHealthPoints(String username) throws RuntimeException {
-    try {
-      //TODO: Fazer uma busca no banco de dados e recuperar o valor de UVHealthPoints baseado no username
-      return 98;
-    } catch (Exception e) {
-      throw new RuntimeException("User not found");
-    }
+  public int findUserHealthPoints(String username) {
+    CosmosService cosmos = new CosmosService();
```
Comment (gabrielsperche) [inline src/main/java/org/splitec/service/UvHealthService.java:51]: @luucasfl  Please add CosmosService as dependency injection to this class

File: src/main/java/org/splitec/service/UvHealthService.java [hunk 9]
```diff
@@ -43,13 +47,10 @@ public int findSecureExposureTime(GetIndex.SafeExposureTime exposureTimeResponse
...
+    User user = cosmos.getUser(username);
```
Comment (gabrielsperche) [inline src/main/java/org/splitec/service/UvHealthService.java:52]: @luucasfl Please abstract this to a method

File: src/main/java/org/splitec/service/CosmosService.java (hunk 7 above)
Comment (luucasfl) [reply src/main/java/org/splitec/service/CosmosService.java:10]: Im not a java super programmer to do this

File: src/main/java/org/splitec/service/UvHealthService.java (hunk 8 above)
Comment (luucasfl) [reply src/main/java/org/splitec/service/UvHealthService.java:51]: @gabrielsperche Im not a Java super programmer to do this

File: src/main/java/org/splitec/client/DatabaseClient.java (hunk 1 above)
Comment (luucasfl) [reply src/main/java/org/splitec/client/DatabaseClient.java:18]: Done

File: src/main/java/org/splitec/client/DatabaseClient.java (hunk 2 above)
Comment (luucasfl) [reply src/main/java/org/splitec/client/DatabaseClient.java:39]: Done

File: src/main/java/org/splitec/client/DatabaseClient.java (hunk 3 above)
Comment (luucasfl) [reply src/main/java/org/splitec/client/DatabaseClient.java:42]: Done

File: src/main/java/org/splitec/client/DatabaseClient.java (hunk 4 above)
Comment (luucasfl) [reply src/main/java/org/splitec/client/DatabaseClient.java:56]: Done

File: src/main/java/org/splitec/client/DatabaseClient.java (hunk 5 above)
Comment (luucasfl) [reply src/main/java/org/splitec/client/DatabaseClient.java:57]: Done

File: src/main/java/org/splitec/client/DatabaseClient.java (hunk 6 above)
Comment (luucasfl) [reply src/main/java/org/splitec/client/DatabaseClient.java:65]: Done

File: src/main/java/org/splitec/service/UvHealthService.java (hunk 8 above)
Comment (luucasfl) [reply src/main/java/org/splitec/service/UvHealthService.java:51]: Done

File: src/main/java/org/splitec/service/UvHealthService.java (hunk 9 above)
Comment (luucasfl) [reply src/main/java/org/splitec/service/UvHealthService.java:52]: Done

**Pull request closed without merging**